unstructured/partition/utils/processing_elements.py

```diff
@@ -0,0 +1,27 @@
+import hashlib
+from collections import defaultdict
+from typing import Tuple
+
+from unstructured_inference.constants import Source
+from unstructured_inference.inference.layout import DocumentLayout
+
+
+def clean_pdfminer_inner_elements(document: DocumentLayout) -> Tuple[DocumentLayout, dict]:
+    """Clean pdfminer elements from inside tables and stores them in extra_info dictionary
+    with the table id as key"""
+    extra_info = defaultdict(list)
+    for page in document.pages:
+        tables = [e for e in page.elements if e.type == "Table"]
+        for i, element in enumerate(page.elements):
+            if element.source == Source.PDFMINER:
+                element_inside_table = [element.bbox.is_in(t.bbox) for t in tables]
+                if sum(element_inside_table) == 1:
+                    parent_table_index = element_inside_table.index(True)
+                    parent_table = tables[parent_table_index]
+                    # Note(Benjamin): is this a good way to guess the id?
+                    future_id = hashlib.sha256(parent_table.text.encode()).hexdigest()[:32]
+                    extra_info[future_id].append(element)
```

Comment:
Should we assume that the element already had an id? I am wondering if it can be reused. Otherwise, I am not sure what should be the best id.

With extra_info, do you mean the removed pdfminer elements that would now be linked to the table?

Reply:
In fact...it doesn't have the id at the moment we're processing this, so I took the code from here:
https://github.com/Unstructured-IO/unstructured/blob/01a0e003d9050a076894aaa4957bd56fe712ea44/unstructured/documents/elements.py#L475-L477
Not sure under which circumstances could fail. 
Probably could try to assign it before in the process 🤔

Reply:
I dig a little bit in the code:
In order for `clean_pdfminer_inner_elements` to work I need a way to store the elements to be possible to recover them in the future (that is why I created that element_id) but at the moment the Element is created by 

https://github.com/Unstructured-IO/unstructured/blob/a5d91e91b9c2961fc8329d63651b7033fe6845f9/unstructured/partition/pdf.py#L439C14-L439C14

the `source` property is deleted (and is the idea), then we have ID but no source to filter.
So, this element_id probably wouldn't be the definitive, but for the moment we don't have other alternatives.